lib/fa/fa_default.hpp

```diff
@@ -28,9 +30,111 @@ namespace fa_default
 	    double best_value;
     };
 
+	struct FaParam
+	{
+	public:
 
-}
+		FaParam(const int, const int, const int, const double, const double, const double,
+				 const double, const double, const double, const double)
+		:T_Max(10000), Particles(100), Dimensions(100), Randomize(0.5), Attract(1.0), Absorb(0.5),
+		 Ran1_Max(1.5), Ran1_Min(-1.5), Ran2_Max(1.5), Ran2_Min(-1.5)
+		{}
+		
+		/* パラメータの定義 */
+    	int T_Max;        // 更新回数
+    	int Particles;    // 個体数
+    	int Dimensions;   // 次元数
+    	double Randomize; // ランダム性パラメータα
+    	double Attract;   // 発光地点における誘引度β
+    	double Absorb;    // 吸収係数γ
+    	double Ran1_Max;  // 初期位置のランダム生成(最大値)
+    	double Ran1_Min;  // 初期位置のランダム生成(最小値)
+    	double Ran2_Max;  // ランダム移動の範囲(最大値)
+    	double Ran2_Min;  // ランダム移動の範囲(最小値)
+	};
+
+	struct FaStrat
+	{
+	protected:
+
+    	double value;                               	   // 最良評価値
+    	std::vector<double> update_value;				   // 各更新時の最良評価値
+    	std::vector<double> variable;                      // 最良評価値の時の変数
+    	std::vector<double> init_variable;                 // 最適化開始時の最良評価値の変数
+    	std::vector<std::vector<double>> update_variable;
+    	double evals;                                      // 評価回数
+    	clock_t time;                                      // 1試行あたりの最適化の実行時間
+    	size_t iter;                                       // 更新回数
```

Comment:
この辺りの基本設計はよし

Reply:
```suggestion
    	double value;                               	   // 最良評価値
    	std::vector<double> update_value;				   // 各更新時の最良評価値
    	std::vector<double> variable;                      // 最良評価値の時の変数
    	std::vector<double> init_variable;                 // 最適化開始時の最良評価値の変数
    	std::vector<std::vector<double>> update_variable;
    	double evals;                                      // 評価回数
    	clock_t start_time;                                      // 1試行あたりの最適化の実行時間
    	size_t iter;                                       // 更新回数
```
念のため、最適化途中の一時的な値ということに注意
最終的に``Result``に渡して返却する